driver/src/solution_submission.rs

```diff
@@ -138,35 +162,17 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
             self.gas_price_estimating,
             60_000_000_000u64.into(),
         )
-        .map_err(|err| {
-            extract_transaction_receipt(&err)
-                .and_then(|tx| {
-                    let block_number = tx.block_number?;
-                    match self
-                        .contract
-                        .get_solution_objective_value(
-                            batch_index,
-                            solution,
-                            Some(block_number.into()),
-                        )
-                        .wait()
-                    {
-                        Ok(_) => None,
-                        Err(e) => Some(SolutionSubmissionError::from(e)),
-                    }
-                })
-                .unwrap_or_else(|| SolutionSubmissionError::Unexpected(err.into()))
-        })
-        .map(|_| ())
+        .or_else(move |err| async move { Err(self.make_error(batch_index, solution, err).await) })
```

Comment:
I was looking around this PR and thinking "I wonder how retry_with_gas_price_increase was handled". Stumbled upon this and might have said the same thing as has already been said. However, in my case I wasn't sure what to expect when I got here. 

Is it safe to assume that the entire content of this error mapping has been offloaded into the cool new `make_error` function? I was kind of under the impression that this particular error was one of a kind and might not generalize well. Do we expect that `make_error` (containing this logic) will be reusable... because currently it is only used here.

Reply:
It is not reusable but it also doesn't have to be reusable. I am splitting a part that can stand alone out of a bigger function.

Reply:
> ubernit: I would find an `async` block with either a `match` on the `retry_with_gas_price_increase` a little easier to understand. 100% OK with how it currently is though.

I don't agree. I feel that if I had written in that way then I would have gotten the comment that `or_else` is specialized method for this and I should use it. I have changed it anyway because everyone else prefers the other way.